Linux64

danoli3 opened the pull request:
Fixes:
- Linux64 
- Linux armv6/armv7/arm64 -

Configure actual build requirement list

danoli3 committed:
Linux GCC8

danoli3 committed:
linux fix build targets

danoli3 committed:
Raspberry Pi 64 - GCC 14.2.0 - Bookworm

danoli3 committed:
Fix download

danoli3 committed:
linux calculate - remove rtAudio

danoli3 committed:
Update toolchain

danoli3 committed:
linux arch

danoli3 committed:
Cross Compilers Linux Raspberry Pi 1 / Zero (armv6), Pi 3A, 3B, 4, Pi 5 (armv7)
Tests for armv6/armv7 enabled

danoli3 committed:
Fix linuxv7 deploy cross compiler

danoli3 committed:
Fix install scripts

danoli3 committed:
Fix install

danoli3 committed:
Cross compiler install extract

danoli3 committed:
fix install

danoli3 committed:
Install and build

danoli3 committed:
Executable

danoli3 committed:
Cross compiling - Configure and set toolchain linked

danoli3 committed:
linux / linux raspberry pi - fmt, libxml2, rtAudio, tess2, zlib

danoli3 committed:
Configure LD path

danoli3 committed:
Linux cross compiler tess2/zlib setup

danoli3 committed:
Cross compiler configuration and debug echo

danoli3 committed:
Cross compiler fix LD

danoli3 committed:
Cross compiler LD flags

danoli3 committed:
GLEW VS arm64/arm64ec fixes exe flags

danoli3 committed:
Glew

danoli3 committed:
Glew linux

danoli3 committed:
Cross configure

danoli3 committed:
Cross

danoli3 committed:
Cross compiler

danoli3 committed:
Linux arm toolchains

danoli3 committed:
Fix backslash in paths

danoli3 committed:
Toolchain root

danoli3 committed:
Toolchain configure

danoli3 committed:
Toolchain fixes

danoli3 committed:
Toolchain formatting

danoli3 committed:
Toolchain update

danoli3 committed:
Toolchain fixes and symlinks for install

danoli3 committed:
GCC prefix

danoli3 committed:
Symlink

danoli3 committed:
rpath

danoli3 committed:
Toolchain fix LD

danoli3 committed:
crt plz

danoli3 committed:
Sysroot

danoli3 committed:
Symlink

danoli3 committed:
link

danoli3 committed:
Toolchain and raspberry pi image

danoli3 committed:
Actions fix Distribution

danoli3 committed:
Raspberry Pi rootfs create fix

dimitre commented:
Hey Dan, I've been using multipass here
https://github.com/dimitre/ofLibs/blob/main/.github/workflows/
take a look on this repo when you have time

danoli3 committed:
Raspbian rootfs download then mount

danoli3 committed:
xz-utils

danoli3 committed:
Fix rootfs

danoli3 committed:
Cleanup toolchains of old

danoli3 committed:
Linux Wayland fix GLFW.
toolchains updated - rpi
added android toolchain.
added linux64 toolchain 64

danoli3 committed:
GLFW fix

danoli3 committed:
GLFW fix - build id 2

danoli3 committed:
Linux GCC 14.
Linux Arm upload action fixed to latest

danoli3 committed:
Fix Github Action

danoli3 committed:
gcc15 from source

danoli3 commented:
> Hey Dan, I've been using multipass here https://github.com/dimitre/ofLibs/blob/main/.github/workflows/ take a look on this repo when you have time

![image](https://github.com/user-attachments/assets/e20a09fd-6c09-47a4-9729-a59999b248e7)

danoli3 committed:
Linux64 test

danoli3 committed:
Linux gcc

danoli3 committed:
Linux rootfs fixes

danoli3 committed:
Actions Checkout v2 -> 4.2.2

danoli3 committed:
Remove fmod calculate

danoli3 committed:
Linux toolchain pass GCC to apothecary

danoli3 committed:
Actions Update

danoli3 committed:
Linux add GCC_VERSION

danoli3 committed:
Linux

danoli3 committed:
Toolchain Linux Fixes

danoli3 committed:
Workflow Optimisations allow disable ENV

danoli3 committed:
Test disable tvOS

danoli3 committed:
Actions disable with if rather than seperate instance of ubuntu

danoli3 committed:
Linux Toolchain fixes

danoli3 committed:
Test Quick Disable

danoli3 committed:
Fix Actions

danoli3 committed:
Workflow

danoli3 committed:
Exit if disabled

danoli3 committed:
linux missing libc

danoli3 committed:
Linux

danoli3 committed:
Arm fixes

danoli3 committed:
Toolchains

danoli3 committed:
Toolchain fix

danoli3 committed:
Chians

danoli3 committed:
Toolchains

danoli3 committed:
Linux sysroot issues

danoli3 committed:
Omg kyle

danoli3 committed:
GLFW fix linux

danoli3 committed:
FMT

danoli3 committed:
Parallel enabled -j cmake

danoli3 committed:
Fix unfinished ssh

danoli3 committed:
Make file fixes

danoli3 committed:
zlib depend

danoli3 committed:
Fix libxml2 depend

danoli3 committed:
LibXML2 defines

danoli3 committed:
libxml2

danoli3 committed:
Parellel make use nproc is not set
libxml2 install
opengl install setup for linux

danoli3 committed:
case sensetive fix

danoli3 committed:
rtAudio linux

danoli3 committed:
rtAudio clean

danoli3 committed:
rtaudio shared to static

danoli3 committed:
Linux GCC 6 disabled
Added Build Script linux
Fix toolchain ripi linux

danoli3 committed:
Enable All workflows

danoli3 committed:
Cross Compiler Setup script ez

danoli3 committed:
Test msys2

danoli3 committed:
Shell on windows fix

danoli3 committed:
svgtiny fix

danoli3 committed:
toolchain linux rpi remove clang

danoli3 committed:
tiny if

danoli3 commented:
Linux 64 compiling all sorted. 
- Fixes the GLFW crash on master due to the package manager version added to distribution 

Also compiles now for GCC14 which allows for C++21.
GCC6 didnt even support c++17.

## Setup builds for 
- GCC11
- GCC12
- GCC13
- GCC14

Added Build script for Linux

Ripi cross compilers almost all setup now for cmake toolchain

danoli3 committed:
Fix linked libraries extra j

danoli3 committed:
Fix depends

danoli3 committed:
svgtiny

danoli3 committed:
svgtiny

danoli3 committed:
Rasbian fixes toolchain pass env if not found

danoli3 committed:
no ccache

danoli3 committed:
fix svgtiny

danoli3 committed:
Fix linux64

danoli3 committed:
svg

danoli3 committed:
svg

danoli3 committed:
Fix linux64 calculate

danoli3 committed:
linux arm64 cross compiler

danoli3 committed:
missing libxrandr-dev for glfw

danoli3 committed:
Linux

danoli3 committed:
fix zlib

danoli3 committed:
Linux build type rework. Github actions linked to
Linux$TYPE now split to using ARCH
Configure scripts moved and relinked

danoli3 commented:
Reworked the Linux$TYPE to be 
Linux target and ARCH as the splitter like we used for other platforms.

## Linux Targets:
- Linux x64 - GCC14 - complete
- Linux arm64 - cross compiler - complete
- Linux Raspberry Pi - armv6 (Pi Zero - 1)  in progress
- Linux Raspberry Pi - armv7 (Pi 2 - 3)  in progress
- Linux Raspberry Piarm64 - aarch64 (Pi 4-5)  in progress
- Nvidia Jetson arm64 (cross compiler in planning)

Since so many linux targets it made sense to split to ABI/ARCH and PLATFORM setting.

danoli3 committed:
Fix apothecary

danoli3 committed:
Build script fix

danoli3 committed:
Fix apothecary

danoli3 committed:
On Target

danoli3 committed:
Jetson cross compiler

danoli3 committed:
Linux fix abi

danoli3 committed:
Fix brotli target

ofTheo commented:
Only thing to be aware of is some fairly recent Linux distros come with GCC 9 / GCC 10 out of the box so if we build libs with the newest GCC it might not be available for a lot of people. 

I think in general we probably want the lowest version of GCC that supports C++17 or whatever other C++ features we are using across the board.

danoli3 committed:
fix brotli

danoli3 committed:
Linux GCC9 / GCC10 enabled

danoli3 committed:
fix gcc not being a number in variable

danoli3 committed:
GCC9

danoli3 committed:
Fix chmod issues scripts - Add linux cross compiler action for arm64 and jetson

danoli3 committed:
zlib linux fix

danoli3 committed:
Chmod fixing sh

danoli3 committed:
apothecary chmod

danoli3 commented:
Good idea - i'll add GCC9 / GCC10 to to the list!


### **GCC 9**
- **Ubuntu**:  
  - Ubuntu 18.04 (via `gcc-9` package from `ppa:ubuntu-toolchain-r/test`).
- **Debian**:  
  - Debian 10 (Buster) offers GCC 9.
- **RHEL/CentOS**:  
  - Available via Devtoolset on RHEL 7.9 and CentOS 7.
- **Fedora**:  
  - Fedora 30 shipped GCC 9 as the default compiler.

---

### **GCC 10**
- **Ubuntu**:  
  - Ubuntu 20.04 (via `gcc-10` package from `ppa:ubuntu-toolchain-r/test`).
- **Debian**:  
  - Debian 11 (Bullseye) includes GCC 10.2.
- **RHEL/CentOS**:  
  - RHEL 8 supports GCC 10 through the GCC Toolset.
- **Fedora**:  
  - Fedora 32 shipped GCC 10 as the system compiler.

---

### **GCC 11**
- **Ubuntu**:  
  - Ubuntu 21.10 and Ubuntu 22.04 ship GCC 11.
- **Debian**:  
  - Debian 12 (Bookworm) uses GCC 11.3.
- **RHEL/CentOS**:  
  - RHEL 9 ships GCC 11 as the default compiler.
- **Fedora**:  
  - Fedora 34 and 35 shipped GCC 11.

---

### **GCC 12**
- **Ubuntu**:  
  - Available via `ppa:ubuntu-toolchain-r/test` for Ubuntu 22.04.
- **Debian**:  
  - Debian 12 includes GCC 12.2.
- **RHEL/CentOS**:  
  - RHEL 9 supports GCC 12 via GCC Toolset.
- **Fedora**:  
  - Fedora 36 shipped GCC 12.

---

### **GCC 13**
- **Ubuntu**:  
  - Available via `ppa:ubuntu-toolchain-r/test` for Ubuntu 22.04 and newer.
- **Debian**:  
  - Not officially included as of 2025 but available via backports for Debian 12.
- **RHEL/CentOS**:  
  - RHEL 9 and 8 support GCC 13 via GCC Toolset.
- **Fedora**:  
  - Fedora 38 ships GCC 13 as the default compiler.

---

### **GCC 14**
- **Ubuntu**:  
  - Shipped with Ubuntu 24.04 (upcoming LTS release).
- **Debian**:  
  - Likely to be included in future Debian 13 (Trixie).
- **RHEL/CentOS**:  
  - Supported via GCC Toolset for RHEL 9.
- **Fedora**:  
  - Fedora 40 ships GCC 14 as the system compiler.
- **Arch Linux**:  
  - As a rolling release, Arch provides the latest stable GCC, currently GCC 14.

---

GCC14  - Shipped with Ubuntu 24.04

danoli3 committed:
zlib

danoli3 committed:
glfw fix

danoli3 committed:
libxml2 linux

danoli3 committed:
Kiss

danoli3 committed:
Linux default sysroot/toolchain /

danoli3 committed:
libxml2 fix copy

danoli3 committed:
Linux Toolchain cross compiler fixes

danoli3 committed:
Kiss kiss

danoli3 committed:
Kiss fix cflags linux

danoli3 committed:
kiss linux

danoli3 committed:
kiss linux line spacing

danoli3 committed:
kiss

danoli3 committed:
libxml2 copy include

danoli3 committed:
svgting

danoli3 committed:
workflow disable test

danoli3 committed:
svgtiny

danoli3 committed:
tess2 fix

danoli3 committed:
fix setup linux

danoli3 committed:
glfw depends

danoli3 committed:
glfw

danoli3 committed:
linux glfw libxcursor-dev

danoli3 committed:
setting up this setup package to work on clean ubuntu

danoli3 committed:
CPP defaults set for GCC versions

danoli3 committed:
brotli fix - configure scripts update

danoli3 committed:
Artifact Cleanups

danoli3 committed:
cross compiler fix

danoli3 committed:
workflows enabled

danoli3 commented:
# Linux 64 
- Updated Linux 64 now using custom Make Toolchain working. 
- Core Libraries with custom fixes and defined in calculate_formulaes for build targets
- Building target library depth changed to /linux/64/libfreeimage.a - where platform / arch location in consistency with rest of targets and compressing of formulae code into single target with configure and toolchain for types 

Renamed Apothecary linuxarmvl linuxarmv6l and linux64 linuxaarch64 to linux. -a is now used for postfix arch type build.
```apothecary/apothecary -t linux -a 64 update core```
```apothecary/apothecary -t linux -a armv6l update core```
```apothecary/apothecary -t linux -a armv7l update core```
```apothecary/apothecary -t linux -a jetson update core```
```apothecary/apothecary -t linux -a aarch64 update core``` - rpi arm64
```apothecary/apothecary -t linux -a arm64 update core``` - native linux arm64

## Linux Cross Compiler Toolchains (in progress)
- Raspberry Pi armv6l, armv7l, aarch64 for Bookworm (with correct Arch targets. Optimizations enabled to maximum for each device for maximum performance including Neon optimisations). -neon -mpu=cortex73 etc
- Linux aarch64 (ARM64) native custom CMake Toolchain added (in progress setup of rootfs install)
- Linux aarch64 (ARM64) Nvidia Jetson custom CMake Toolchain added setup, install, building

danoli3 commented:
Added Github Action env.Variable to disable / enable the actual building of other formulae
```  DISABLE_WORKFLOW: "false" ```

Will use this instead of moving to disabled

danoli3 commented:
Will progress with Linux Toolchains also updating Android Toolchain to Latest NDK.

Lets get this Linux stuff in and fix issues with path changes in core @dimitre / project Generator I think it could break some path depth stuff but will be worth fixing.

Mering now. LEeerroy

danoli3 commented:
Okay awesome yeah linux libs havent uploaded through yet package not added to release shall fix up tomorrow but all platforms test with updated code
